Source/path.h

```diff
@@ -28,23 +28,49 @@ struct PATHNODE {
 
 bool IsTileNotSolid(Point position);
 bool IsTileSolid(Point position);
+
+/**
+ * @brief Checks the position is solid or blocked by an object
+ */
 bool IsTileWalkable(Point position, bool ignoreDoors = false);
-int FindPath(const std::function<bool(Point)> &posOk, int sx, int sy, int dx, int dy, int8_t path[MAX_PATH_LENGTH]);
-int path_get_h_cost(int sx, int sy, int dx, int dy);
-PATHNODE *GetNextPath();
-bool path_solid_pieces(PATHNODE *pPath, int dx, int dy);
-bool path_get_path(const std::function<bool(Point)> &posOk, PATHNODE *pPath, int x, int y);
-bool path_parent_path(PATHNODE *pPath, int dx, int dy, int sx, int sy);
-PATHNODE *path_get_node1(int dx, int dy);
-PATHNODE *path_get_node2(int dx, int dy);
-void path_next_node(PATHNODE *pPath);
-void path_set_coords(PATHNODE *pPath);
-void path_push_active_step(PATHNODE *pPath);
-PATHNODE *path_pop_active_step();
-PATHNODE *path_new_step();
-
-/* rdata */
-
-extern const Displacement PathDirs[8];
+
+/**
+ * @brief Find the shortest path from startPosition to destinationPosition, using PosOk(Point) to check that each step is a valid position.
+ * Store the step directions (corresponds to an index in PathDirs) in path, which must have room for 24 steps
+ */
+int FindPath(const std::function<bool(Point)> &posOk, Point start, Point destination, int8_t path[MAX_PATH_LENGTH]);
+
+/**
+ * @brief check if stepping from a given position to a neighbouring tile cuts a corner.
+ *
+ * If you step from A to B, both Xs need to be clear:
+ *
+ *  AX
+ *  XB
+ *
+ * @return true if step is allowed
+ */
+bool path_solid_pieces(Point sourcePosition, Point destinationPosition);
+
+/**
+ * @brief heuristic, estimated cost from sourcePosition to destinationPosition.
+ *
+ * This is an internal function that is only exposed to allow for testing the way path weights are determined.
+ */
+int path_get_h_cost(Point sourcePosition, Point destinationPosition);
+
+/** For iterating over the 8 possible movement directions */
+const Displacement PathDirs[8] = {
+	// clang-format off
+	{ -1, -1 }, //DIR_N
+	{ -1,  1 }, //DIR_W
+	{  1, -1 }, //DIR_E
+	{  1,  1 }, //DIR_S
+	{ -1,  0 }, //DIR_NW
+	{  0, -1 }, //DIR_NE
+	{  1,  0 }, //DIR_SE
+	{  0,  1 }, //DIR_SW

```

Comment:
Any reason not to convert this to an array of directions now?

Reply:
FindMeleeTarget uses it in a rather annoying way, but mostly I was trying to work out if the order mattered so it could be replaced with one of the other collections of `Direction` enum values excluding DIR_OMNI. Didn't want to get too deep into that refactoring in this PR.